### ISSUE TITLE
More details about the results on Appendix B.4 

### ISSUE THREAD

**Issue by LoverLost** [boyiwei/alignment-attribution-code#3]
In Appendix B.4 of your paper, you analyzed the accuracy of linear probers trained on all 1024 attention heads from Llama2-7B-chat  and1600 attention heads from Llama2-13B-chat. I would like to ask how this accuracy is calculated?I mean, how do you perform the steps to evaluate each attention head?

**Comment by Unispac** [comment]
Hi, thank your for your question!
As noted in our paper:

"""
We adopt a similar probing setup used in [Li et al. (2023a)](https://arxiv.org/abs/2306.03341) for identifying safety-critical neurons in
this work. Specifically, we feed the model with all 420 harmful instructions from AdvBench, as well as 420 harmless
instructions randomly sampled from the utility dataset (more specifically, Alpaca dataset). We collect the activation outputs of every internal attention head for these instructions. This collected data is then split into two sets, with a 5 : 2 ratio for the training split and the validation split, respectively. For each attention head, we train a linear classifier on the training split using its activation inputs to distinguish between activations resulting from harmful and harmless instructions. We then evaluate the accuracy of the classifier on the validation split, which indicates the relevance of the attention head in distinguishing between harmful and harmless instructions.
"""

So basically, for each of the attention head, we only take the output features of that attention head, using those features for both training the probe classifier and evaluate the accuracy of the classifier.
